lib/engine.js

```diff
@@ -393,52 +393,51 @@ class Engine {
-            transformerInput = await getSource(
-                input.url,
-                transformerDirectory.base,
-                false
-            );
-
-            this[INTERNAL].timers.download.stop();
-
-        } // url to asset is available and transformer to use it to process asset (ex: worker-ffmpeg)
-        else if(!downloadAsset && input.url) {
-            // dont download - T.compute will use url to process asset
-            // input.url is already set and is passed on to Asset object 
-            debug(`asset url available and disableInputDownload is ${disableInputDownload} : dont need to do anything`);
-            transformerInput = new Asset(input);
-        } // if (file exists locally && (T.manifest.input.sourceType === 'URL') then upload to tmp cloud
-        else if(!downloadAsset && input.path) {
-            const preSignedUrl = await this.getPresignedUrl(input.path);
-            // TODO: remove this log
-            debug('generated presigned url', preSignedUrl);
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) {
+            throw new GenericError('No source file accessible.');
         }
 
-        if (!transformerInput || !transformerInput.path) {
-            throw new GenericError('No source file accessible.');
+        debug(`sourceType: ${input.sourceType}`);
+        if (input.sourceType === TransformerSourceType.URL) {
+            const inputUrl =  input.url || '';
+            if (input.url && validUrl.isHttpsUri(inputUrl)) {
+                debug(`asset url available : dont need to do anything`);
+                transformerInput = new Asset(input);
+            } else if (input.url && isDataUrlAndValid(inputUrl)) {
+                debug(`asset url available as data URI: upload asset to a temp cloud storage and generate a presigned URL`);
+                this[INTERNAL].timers.upload.start();
+                transformerInput = await getSource(input, transformerDirectory.base, true);
+                this[INTERNAL].timers.upload.stop();
+            } else {
+                debug(`asset url not available: generated presigned url`);
+                try {
+                    this[INTERNAL].timers.upload.start();
+                    const preSignedUrl = await this.getPresignedUrl(input.path);
+                    this[INTERNAL].timers.upload.stop();
+                    input.url = preSignedUrl;
+                    transformerInput = new Asset(input, input.path);
+                } catch (error) {
+                    console.log(error);
+                    throw new GenericError(`Fail generating presigned url.`);
+                }
+            }
+        } else { // LOCAL otherwise
```

Comment:
please add a note,
defaults to assuming the asset is needed locally

or we can explicitely check it is `LOCAL` and throw if its undefined

Reply:
done

Reply:
default to LOCAL as we don't always specific sourceType in test, I am not sure if that is also the case in execution